src/Algebra/Graph/AdjacencyIntMap/Algorithm.hs

```diff
@@ -122,15 +98,15 @@ bfsForestFrom' vs g = evalState (explore vs) IntSet.empty where
 -- bfs [2] ('edge' 1 2)                                   == [[2]]
 -- bfs [3] ('edge' 1 2)                                   == []
 -- bfs [1,2] ('edge' 1 2)                                 == [[1],[2]]
--- bfs [2,1] ('edge' 1 2)                                 == [[2],[1]]
+-- bfs [2,1] ('edge' 1 2)                                 == [[2,1]]
 -- bfs [1,2] ( (1*2) + (3*4) + (5*6) )                  == [[1],[2]]
--- bfs [1,3] ( (1*2) + (3*4) + (5*6) )                  == [[1],[2],[3],[4]]
+-- bfs [1,3] ( (1*2) + (3*4) + (5*6) )                  == [[1,3],[2,4]]
 -- bfs [3] (3 * (1 + 4) * (1 + 5))                      == [[3],[1,4,5]]
 -- bfs [2] ('circuit' [1..5] + 'circuit' [5,4..1])          == [[2],[1,3],[5,4]]
 -- 'concat' (bfs [3] $ 'circuit' [1..5] + 'circuit' [5,4..1]) == [3,2,4,1,5]
 -- @
 bfs :: [Int] -> AdjacencyIntMap -> [[Int]]
-bfs vs = bfsForestFrom vs >=> levels
+bfs vs = map concat . List.transpose . map levels . bfsForest vs
```

Comment:
This is nice brain teaser :) Perhaps, it's worth documenting it as a property, since it's far from obvious for me.

Reply:
Haha yes, the realization came as a modest light-bulb moment. I'm not sure I understand what it means to document as a property though?